boards/PDM/Inc/Io/Io_Efuse.h

```diff
@@ -0,0 +1,726 @@
+/**
+ * @brief Driver for interfacing with the 22XS4200 E-Fuse via SPI
+ * @Settings Efuse max SPI transfer rate: 8MHz
+ *           SPI Clock Polarity: 0 (SCK low-level idle state)
+ *           SPI Clock Phase: 2nd edge (slave samples MOSI on SCK falling edge)
+ *           Slave Select: Active Low (Must be toggled between SPI messages)
+ */
+#pragma once
+
+#include <stdint.h>
+#include <stdbool.h>
+#include "main.h"
+
+// Current Sense Ratios
+#define HIGH_CURRENT_SENSE_RATIO (1.0f / 1500.0f)
+#define LOW_CURRENT_SENSE_RATIO (1.0f / 500.0f)
+
+// Current Sense Channels
+#define AUX1_CURRENT_SENSE_CHANNEL CSNS_FUNCTION_CURRENT_CH0
+#define AUX2_CURRENT_SENSE_CHANNEL CSNS_FUNCTION_CURRENT_CH1
+#define EM_CURRENT_SENSE_CHANNEL CSNS_FUNCTION_CURRENT_CH0
+#define PUMP_CURRENT_SENSE_CHANNEL CSNS_FUNCTION_CURRENT_CH1
+#define DRIVE_INVERTER_L_CURRENT_SENSE_CHANNEL CSNS_FUNCTION_CURRENT_CH0
+#define DRIVE_INVERTER_R_CURRENT_SENSE_CHANNEL CSNS_FUNCTION_CURRENT_CH1
+#define AIR_SHDN_CURRENT_SENSE_CHANNEL CSNS_FUNCTION_CURRENT_CH0
+#define GLV_CURRENT_SENSE_CHANNEL CSNS_FUNCTION_CURRENT_CH1
+
+//! Serial input register addresses
+
+// STATR_s input address
+#define SI_STATR_0_ADDR 0x00U
```

Comment:
I still don't see how most of these defines must be in the header? I imagined only the functions need to be made public, not all these macros.